include/swift/SIL/SILBuilder.h

```diff
@@ -260,8 +303,10 @@ class SILBuilder {
   ApplyInst *createApply(SILLocation Loc, SILValue Fn, SILType SubstFnTy,
                          SILType Result, ArrayRef<Substitution> Subs,
                          ArrayRef<SILValue> Args, bool isNonThrowing) {
-    return insert(ApplyInst::create(getSILDebugLocation(Loc), Fn, SubstFnTy,
+    auto *I = insert(ApplyInst::create(getSILDebugLocation(Loc), Fn, SubstFnTy,
                                     Result, Subs, Args, isNonThrowing, F));
+    createApplyTypeDefs(I, Subs);
+    return I;
```

Comment:
I feel pretty strongly that this belongs in ApplyInst::create, especially since you have to allocate the right amount of space for the extra operands there anyway.  If you want to track the opening sites of instructions in SILBuilder rather than the SILFunction, okay, but that just means that you should pass down the tracker, not that you should make SILBuilder actually responsible for completing the initialization of the instruction.